DQMOffline/Trigger/plugins/MssmHbbMonitor.cc

```diff
@@ -0,0 +1,203 @@
+  dphi_jet12_ = ibooker.book1D("dphi_jet12","dphi_jet12",jetPhibins_.nbins,jetPhibins_.xmin,jetPhibins_.xmax);
+  dr_jet12_   = ibooker.book1D("dr_jet12","dr_jet12",jetDRbins_.nbins,jetDRbins_.xmin,jetDRbins_.xmax);
+
+  discr_offline_btag_jet1_ = ibooker.book1D("discr_offline_btag_jet1","discr_offline_btag_jet1",jetBtagbins_.nbins,jetBtagbins_.xmin,jetBtagbins_.xmax);
+  discr_offline_btag_jet2_ = ibooker.book1D("discr_offline_btag_jet2","discr_offline_btag_jet2",jetBtagbins_.nbins,jetBtagbins_.xmin,jetBtagbins_.xmax);
+
+  pt_muon_  = ibooker.book1D("pt_muon","pt_muon",muonPtbins_.nbins,muonPtbins_.xmin,muonPtbins_.xmax);
+  eta_muon_ = ibooker.book1D("eta_muon","eta_muon",muonEtabins_.nbins,muonEtabins_.xmin,muonEtabins_.xmax);
+  phi_muon_ = ibooker.book1D("phi_muon","phi_muon",muonPhibins_.nbins,muonPhibins_.xmin,muonPhibins_.xmax);
+    
+  // Initialize the GenericTriggerEventFlag
+}
+
+void MssmHbbMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)
+{
+
+   bool accept = false;
+   
+   bool doMuon = true;
+   
+   edm::Handle<edm::TriggerResults> triggerResultsHandler;
+   iEvent.getByToken(triggerResultsToken_, triggerResultsHandler);
+   if ( ! triggerResultsHandler.isValid() )  return;
+      
+   edm::Handle<reco::JetTagCollection> offlineJetTagPFHandler;
+   iEvent.getByToken(offlineBtagToken_, offlineJetTagPFHandler);      
+   if( ! offlineJetTagPFHandler.isValid()) return;
+   
+   edm::Handle<reco::MuonCollection> muonHandler;
+   iEvent.getByToken(muonsToken_, muonHandler);
+   if( ! muonHandler.isValid()) doMuon = false;
+   
+   const edm::TriggerResults & triggers = *(triggerResultsHandler.product());
+   
+   for ( size_t j = 0 ; j < hltConfig_.size() ; ++j )
+   {
+      if ( hltConfig_.triggerName(j).find(pathname_) == 0 )
+      {
+         if ( triggers.accept(j) ) 
+         {
+            accept = true;
+            break;
+         }
+      }
+   }
+   
+   if ( accept )
+   {
+      
+      reco::JetTagCollection jettags = *(offlineJetTagPFHandler.product());
+      if ( jettags.size() > 1 )
+      {
+         const reco::Jet * jet1 = jettags.key(0).get();
+         const reco::Jet * jet2 = jettags.key(1).get();
+         TLorentzVector p4_jet1;
+         p4_jet1.SetPtEtaPhiM(jet1->pt(),jet1->eta(),jet1->phi(),0);
+         TLorentzVector p4_jet2;
+         p4_jet2.SetPtEtaPhiM(jet2->pt(),jet2->eta(),jet2->phi(),0);
+         float btag1 = jettags.value(0);
+         float btag2 = jettags.value(1);
+         if ( btag1 < 0 ) btag1 = -0.5;
+         if ( btag2 < 0 ) btag2 = -0.5;
+         
+         float deta12 = p4_jet1.Eta() - p4_jet2.Eta();
+         float dphi12 = p4_jet1.DeltaPhi(p4_jet2);
+         float dr12   = p4_jet1.DeltaR(p4_jet2);
+            
+         if ( jet1->pt() > jetPtmin_ && jet2->pt() > jetPtmin_ && fabs(jet1->eta()) < jetEtamax_ && fabs(jet2->eta()) < jetEtamax_ )
```

Comment:
why don't you apply such selection before ?
you could even save some memory and cpu, couldn't you ?